h/test/app_test.py

```diff
@@ -0,0 +1,70 @@
+# -*- coding: utf-8 -*-
+
+import mock
+import pytest
+
+from pyramid.testing import DummyRequest
+
+from h import app
+
+class TestPublishAfterCommitEvents(object):
+    def test_skips_publishing_events_on_exception(self):
+        request = self.makerequest(exception=ValueError('exploded!'))
+        app.publish_after_commit_events(request, None)
+        assert not request.after_commit_events.publish_all.called
+
+    def test_skips_publishing_events_when_no_events_queued(self):
+        request = self.makerequest(events=[])
+        app.publish_after_commit_events(request, None)
+        assert not request.after_commit_events.publish_all.called
+
+    def test_publishes_events(self):
+        request = self.makerequest(events=[mock.Mock()])
+        app.publish_after_commit_events(request, None)
+        assert request.after_commit_events.publish_all.called
+
+    def makerequest(self, exception=None, events=None):
```

Comment:
I've been using the name `mock_request()` for test helper functions that return request objects elsewhere, might be good to keep the naming style consistent